bottlecap/src/proxy/interceptor.rs

```diff
@@ -153,6 +160,18 @@ async fn invocation_next_proxy(
             }
         };
 
+    if let Some(appsec) = appsec_processor {
+        if let Some(rid) = intercepted_parts
+            .headers
+            .get("Lambda-Runtime-Aws-Request-Id")
+            .and_then(|v| v.to_str().ok())
+        {
+            if let Ok(trigger) = IdentifiedTrigger::from_slice(&intercepted_bytes) {
```

Comment:
What happens if no trigger is found, is there a chance that you still receive elements that are worth checking for ASM?

Was this the same way as the Go counterpart?

Reply:
If we don't have an identified trigger, then there is nothing for us to look at... We are interested in the HTTP payload's content, which requires we can parse the lambda input.

This is the same as is done in Go today (no identified trigger => nothing done).

Reply:
I see, I wonder what happens in the use case where a customer invokes the lambda directly using the AWS SDK, we can go through this edge case later, but AFAIK, it doesnt get identified as any trigger, yet it does receive HTTP headers and other metadata

Reply:
Yah if it's not an AppSec-relevant payload there isn't much we can do with it...